subprojects/jacoco/src/integTest/groovy/org/gradle/testing/jacoco/plugins/JacocoAggregationIntegrationTest.groovy

```diff
@@ -0,0 +1,486 @@
+                        testType = TestType.INTEGRATION_TESTS
+                        useJUnit()
+                        dependencies {
+                            implementation project(':transitive') // necessary to access Divisor when compiling test
+                        }
+                    }
+                }
+            }
+        """
+
+        file("transitive/src/main/java/transitive/Divisor.java").java """
+                package transitive;
+
+                public class Divisor {
+                    public int div(int x, int y) {
+                        return x / y;
+                    }
+                    public int mod(int x, int y) {
+                        return x % y;
+                    }
+                }
+            """
+
+        file("application/src/integTest/java/application/DivTest.java").java """
+                package application;
+
+                import org.junit.Assert;
+                import org.junit.Test;
+                import transitive.Divisor;
+
+                public class DivTest {
+                    @Test
+                    public void testDiv() {
+                        Divisor divisor = new Divisor();
+                        Assert.assertEquals(2, divisor.div(4, 2));
+                    }
+                }
+            """
+        file("transitive/src/integTest/java/transitive/ModTest.java").java """
+                package transitive;
+
+                import org.junit.Assert;
+                import org.junit.Test;
+
+                public class ModTest {
+                    @Test
+                    public void testMod() {
+                        Divisor divisor = new Divisor();
+                        Assert.assertEquals(1, divisor.mod(5, 2));
+                    }
+                }
+            """
+
+        when:
+        succeeds(":application:testCodeCoverageReport", ":application:integTestCodeCoverageReport")
+
+        then:
+        file("transitive/build/jacoco/test.exec").assertExists()
+        file("transitive/build/jacoco/integTest.exec").assertExists()
+        file("direct/build/jacoco/test.exec").assertExists()
+        file("direct/build/jacoco/integTest.exec").assertDoesNotExist()
+        file("application/build/jacoco/test.exec").assertExists()
+        file("application/build/jacoco/integTest.exec").assertExists()
+
+        file("application/build/reports/jacoco/testCodeCoverageReport/html/index.html").assertExists()
+        file("application/build/reports/jacoco/integTestCodeCoverageReport/html/index.html").assertExists()
+
+        def testReport = new JacocoReportXmlFixture(file("application/build/reports/jacoco/testCodeCoverageReport/testCodeCoverageReport.xml"))
+        testReport.assertHasClassCoverage("application.Adder")
+        testReport.assertHasClassCoverage("direct.Multiplier")
+        testReport.assertHasClassCoverage("transitive.Powerize")
+        testReport.assertHasClassCoverage("transitive.Divisor", 0)
```

Comment:
Maybe a `assertLacksClassCoverage()` or `assertFoundButNoCoverage` or something instead of passing the 0?  I think that looks a little odd.

Reply:
Good idea